frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx

```diff
@@ -518,4 +518,10 @@ function isSuccessfulFetchDashboardResult(
   return !hasError;
 }
 
+export function isCancelledFetchDashboardResult(
+  result: FetchDashboardResult,
+): result is CancelledFetchDashboardResult {
+  return !!(result as CancelledFetchDashboardResult)?.payload?.isCancelled;
```

Comment:
My suggestion was to avoid the cast by introducing a type guard, but the cast is still here.

```suggestion
  return isObject(result.payload) && Boolean(result.payload.isCancelled);
```

We have [`isObject` already](https://github.com/metabase/metabase/blob/9839606a1bdeb6d2680f69a590b957ddb05432fd/frontend/src/metabase-types/guards/click-behavior.ts#L9-L13), we're going to need to move it somewhere else and reuse it here.